Previous PR to fix backup warning causes crash

### What happened?

PR #1470 works when testing via `gradlew runClient`, but leads to a crash with the built JAR.
I've already made a PR to revert this temporarily (#1476) which has already been approved, so that the beta branch is usable again.

This has been first reported by @Addi3 on Discord and I've already reproduced it.
I've self-assigned the issue and will investigate how to fix it, keeping this issue for tracking purposes.

Steps to reproduce the behavior (in :
1. Start the game, try to open a new map.
2. See error

### Version

1.2.0

### Build number & branch

689

### Logs & screenshots

The pertinent part:

<details>

```
Caused by: org.spongepowered.asm.mixin.injection.throwables.InjectionError: Critical injection failure: Callback method skipBackupScreen(Lnet/minecraft/class_437;Ljava/lang/String;Lorg/spongepowered/asm/mixin/injection/callback/CallbackInfo;)V in ait.mixins.json:client.experimental_screen.WorldOpenFlowsMixin from mod ait failed injection check, (0/1) succeeded. Scanned 0 target(s). Using refmap ait-refmap.json
    at org.spongepowered.asm.mixin.injection.struct.InjectionInfo.postInject(InjectionInfo.java:531)
    at org.spongepowered.asm.mixin.transformer.MixinTargetContext.applyInjections(MixinTargetContext.java:1490)
    at org.spongepowered.asm.mixin.transformer.MixinApplicatorStandard.applyInjections(MixinApplicatorStandard.java:752)
    at org.spongepowered.asm.mixin.transformer.MixinApplicatorStandard.applyMixin(MixinApplicatorStandard.java:330)
    at org.spongepowered.asm.mixin.transformer.MixinApplicatorStandard.apply(MixinApplicatorStandard.java:246)
    at org.spongepowered.asm.mixin.transformer.TargetClassContext.apply(TargetClassContext.java:437)
    at org.spongepowered.asm.mixin.transformer.TargetClassContext.applyMixins(TargetClassContext.java:418)
    at org.spongepowered.asm.mixin.transformer.MixinProcessor.applyMixins(MixinProcessor.java:363)
    ... 36 more
```
</details>

### I've checked the reported issues...

- [x] ...and this issue was not reported before.

--- a/src/main/java/dev/amble/ait/mixin/client/experimental_screen/WorldOpenFlowsMixin.java
+++ b/src/main/java/dev/amble/ait/mixin/client/experimental_screen/WorldOpenFlowsMixin.java
@@ -1,25 +1,25 @@
 package dev.amble.ait.mixin.client.experimental_screen;
 
-import com.mojang.serialization.Lifecycle;
+import net.minecraft.client.gui.screen.Screen;
 import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.Shadow;
 import org.spongepowered.asm.mixin.injection.At;
 import org.spongepowered.asm.mixin.injection.Inject;
 import org.spongepowered.asm.mixin.injection.callback.CallbackInfo;
 
-import net.minecraft.client.MinecraftClient;
-import net.minecraft.client.gui.screen.world.CreateWorldScreen;
 import net.minecraft.server.integrated.IntegratedServerLoader;
 
 import dev.amble.ait.AITMod;
 
 @Mixin(value = IntegratedServerLoader.class)
-public class WorldOpenFlowsMixin {
+public abstract class WorldOpenFlowsMixin {
 
-    @Inject(method = "tryLoad", at = @At(value = "INVOKE_ASSIGN", target = "Lcom/mojang/serialization/Lifecycle;experimental()Lcom/mojang/serialization/Lifecycle;", remap = false), cancellable = true)
-    private static void confirmWorldCreation(MinecraftClient client, CreateWorldScreen parent, Lifecycle lifecycle,
-            Runnable loader, boolean bypassWarnings, CallbackInfo ci) {
+    @Shadow protected abstract void start(Screen parent, String levelName, boolean safeMode, boolean canShowBackupPrompt);
+
+    @Inject(method = "start(Lnet/minecraft/client/gui/screen/Screen;Ljava/lang/String;)V", at = @At(value = "INVOKE", target = "Lnet/minecraft/server/integrated/IntegratedServerLoader;start(Lnet/minecraft/client/gui/screen/Screen;Ljava/lang/String;ZZ)V"), cancellable = true)
+    private void skipBackupScreen(Screen parent, String levelName, CallbackInfo ci) {
         if (!AITMod.CONFIG.CLIENT.SHOW_EXPERIMENTAL_WARNING) {
-            loader.run();
+            this.start(parent, levelName, false, false);
             ci.cancel();
         }
     }

